openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -23,202 +23,114 @@
-                if _.get("canonicalName") is not None
-            ]
-            taxa_list = [_.get("string", "") for _ in taxa]
-
-        if taxa_list:
-            title = taxa_list[0]
-
-        return taxa_list, title
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        """
+        Get the data for a single image record.
 
-    @staticmethod
-    def _get_creator_details(result: dict) -> tuple[str | None, str | None, str | None]:
-        credit_line = None
-        pub_date = None
-        creator = None
-        submitter = result.get("submitter", {})
-        first_name = submitter.get("firstName")
-        last_name = submitter.get("lastName")
-        if first_name and last_name:
-            creator = f"{first_name} {last_name}".strip()
-
-        if credit := result.get("credit"):
-            credit_line = credit.strip()
-            pub_date = result.get("submitted").strip()
-
-        return creator, credit_line, pub_date
+        TODO: Adapt `url` and `creator_url` to avoid redirects.
+        """
 
-    def get_record_data(self, data: dict) -> dict | list[dict] | None:
-        uid = data.get("uid")
+        uid = data.get("uuid")
         if not uid:
-            return
-        logger.debug(f"Processing UUID: {uid}")
-        params = {
-            "options": " ".join(
-                [
-                    "credit",
-                    "licenseURL",
-                    "pngFiles",
-                    "submitted",
-                    "submitter",
-                    "taxa",
-                    "canonicalName",
-                    "string",
-                    "firstName",
-                    "lastName",
-                ]
-            )
-        }
-        endpoint = f"{self.base_endpoint}/{uid}"
-        response_json = self.get_response_json(params, endpoint)
-        result = self._get_response_data(response_json)
-        if not result:
             return None
 
-        meta_data = {}
-        uid = result.get("uid")
-        license_url = result.get("licenseURL")
-
-        img_url, width, height = self._get_image_info(result, uid)
-
-        if img_url is None:
+        data = data.get("_links", {})
+        license_url = data.get("license", {}).get("href")
+        img_url = data.get("sourceFile", {}).get("href")
+        foreign_url = data.get("self", {}).get("href")
+        if not license_url or not img_url or not foreign_url:
             return None
 
-        meta_data["taxa"], title = self._get_taxa_details(result)
+        foreign_url = self.host + foreign_url
 
-        foreign_url = self._image_url(uid)
-
-        (
-            creator,
-            meta_data["credit_line"],
-            meta_data["pub_date"],
-        ) = self._get_creator_details(result)
+        title = data.get("self", {}).get("title")
+        creator = data.get("contributor", {}).get("title")
+        creator_url = self.host + data.get("contributor", {}).get("href")
```

Comment:
This could return an invalid url if `contributor` is `None`
```suggestion
        creator_path = data.get("contributor", {}).get("href")
        creator_url = self.host + creator_path if creator_path else None
```